man/systemd.socket.xml

```diff
@@ -377,7 +377,12 @@
         sockets and FIFO nodes in the file system are owned by the specified user and group. If unset (the
         default), the nodes are owned by the root user/group (if run in system context) or the invoking
         user/group (if run in user context).  If only a user is specified but no group, then the group is
-        derived from the user's default group.</para>
+        derived from the user's default group.
+        Note that this might not have the desired effect if a socket happens to be inside a directory also
+        referred to in a <varname>RuntimeDirectory=</varname>, <varname>StateDirectory=</varname>,
+        <varname>CacheDirectory=</varname>, or <varname>LogsDirectory=</varname> of any service, due to the
+        change of ownership caused by these options.
```

Comment:
I am not following? What's the problem?

Reply:
As in the issue description, `SocketUser=` is used to make the socket file owned by a different user than the service. If you however also have `RuntimeDirectory=` set, and your socket file is in there, both places try to chown this.

The common example for breakage is `myservice` wanting a `/run/myservice/sock` to be owned by `otheruser`.

If `myservice` also sets `RuntimeDirectory=myservice`, which chowns to `myservice` user, these two things race.